### DISCUSSION TITLE
Feasibility of using DeckGL for indoor maps and schematics

### DISCUSSION THREAD

**Discussion by jasonsturges** [visgl/deck.gl#6156] [Q&A] [answered]
Currently leveraging Pixi.js and Fabric.js for indoor mapping and schematic projects; however, I'm taken with DeckGL on multiple fronts and curious whether it would be feasible to consider for these projects.

Example: dynamically edited via Fabric.js, runtime viewed in Pixi.js:
![schematic](https://user-images.githubusercontent.com/1213591/131785460-e66647db-445e-47bd-99f8-1235f7ef9012.gif)

Following the Icon Layer example, I've prototyped using a `BitmapLayer` background with I/O devices as icons within an `IconLayer`, and DeckGL instance as `OrthographicView`.

Is this too far outside the bounds of DeckGL intended use?

If feasible, is there a better approach than the icon layer?

**Comment by ibgreen** [comment, marked as answer]
> Is this too far outside the bounds of DeckGL intended use?

We welcome use of deck.gl in new areas (as you can see in our showcases on the website, deck.gl is gaining momentum in biological visualizations with some very high caliber work being done, custom loaders and layers etc), however you may be the first using deck.gl for this use case. If you run into special requirements you may need to be ready to roll up your sleeves.

> If feasible, is there a better approach than the icon layer?

Another approach would be to render each element as graphics. While custom layers are of course an options, we do have a ScenegraphLayer that can render glTF files. They are normally 3D but could also contain 2D graphics. You'd need to create the glTF files yourself in blender or another appropriate tools.

With the CompositeLayer you can also build up visualizations from primitive layers (lines, points, polygons etc).

You are welcome to add your visualization to our showcases. It might help drum up more interest in using deck.gl for your use case.
